rtgui/editcallbacks.h

```diff
@@ -0,0 +1,256 @@
+/*
+ *  This file is part of RawTherapee.
+ *
+ *  Copyright (c) 2019 Jean-Christophe FRISCH <natureh.510@gmail.com>
+ *
+ *  RawTherapee is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  RawTherapee is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with RawTherapee.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "editid.h"
+#include "cursormanager.h"
+#include "../rtengine/coord.h"
+
+class Geometry;
+class EditDataProvider;
+
+/** @file
+ * See editwidgets.h for documentation
+ */
+
+/// @brief Method for client tools needing Edit information
+class EditSubscriber
+{
+
+public:
+
+private:
+    EditUniqueID ID; /// this will be used in improcfun to locate the data that has to be stored in the buffer; it must be unique in RT
+    EditType editingType;
+    BufferType bufferType;
+    EditDataProvider *provider;
+
+protected:
+    std::vector<Geometry*> visibleGeometry;    /// displayed geometry
+    std::vector<Geometry*> mouseOverGeometry;  /// mouseOver geometry, drawn in a hidden buffer
+    enum {
+        ES_ACTION_NONE,      ///
+        ES_ACTION_DRAGGING,  /// set action to this value in the buttonPressed event to start dragging and ask for drag event
+        ES_ACTION_PICKING    /// set action to this value in the buttonPressed event whenever the user is picking something through a single click. In this case, the pickX events will be called INSTEAD of buttonXReleased !
+    } action;                /// object mode only, ignored in Pipette mode
+
+public:
+    explicit EditSubscriber (EditType editType);
+    virtual ~EditSubscriber () {}
+
+    void               setEditProvider(EditDataProvider *provider);

```

Comment:
Might be unneeded if passing through c'tor is possible.

Reply:
Already answered above.